src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.cs

```
@@ -554,6 +554,10 @@ internal List<ControlInformation> GetControlsInformation()
                         {
                             controlInfo.Name = prop.GetValue(c);
                         }
+                        else
+                        {
+                            controlInfo.Name = string.Empty;
```

Why should not serialization throw for an invalid control?

---

This throws a NullReferenceException, which is better handled than thrown. If you look at the Netfx code, we debug assert - this behaviour is not expected. I would rather we handled serialising and deserialising preserving state than throwing here.

---

I don't know if its relevant here, but controls without a name are (in general) not invalid. When generating controls at runtime using code (instead of design time through the VS designer) you often don't set a name. In general, controls having a name is not a checked requirement in WinForms.

---

In the case of design time serialization, control with an empty string instead of a name is an invalid control. This change will cause failure later when the serialized content is processed, it's more convenient to debug an explicit null reference exception in serialization code.

---

> This change will cause failure later when the serialized content is processed
Not true actually, I think it will just be ignored

---

It will be ignored and this is exactly the problem because we will get a table layout panel with an empty cell. This bug will be more difficult to understand or even notice.

Empty string is as invalid as a null is because such controls are not deserialized from the XML by TypeDescriptor.ConvertFrom  - https://github.com/dotnet/winforms/blob/a53da71400af51593036bb37cb802de80915ed60/src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettingsTypeConverter.cs#L153-L173